lib/ruby_llm/providers/bedrock/models.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module RubyLLM
+  module Providers
+    module Bedrock
+      # Models methods for the AWS Bedrock API implementation
+      module Models
+        def list_models
+          @connection = nil # reset connection since base url is different
+          @api_base = "https://bedrock.#{aws_region}.amazonaws.com"
+          response = connection.get(models_url) do |req|
+            req.headers.merge! headers(method: :get,
+                                       path: "#{connection.url_prefix}#{models_url}")
+          end
+          @connection = nil # reset connection since base url is different
+
+          parse_list_models_response(response, slug, capabilities)
+        end
```

Comment:
Is it necessary to override `list_models` in light of the comment I added above for `post` and `headers`?